src/lib/components/VideoCard/VideoCard.svelte

```diff
@@ -1,78 +1,151 @@
 <script lang="ts">
   import "./VideoCard.scss";
 
-  import { afterUpdate } from "svelte";
-
-  import getYoutubeVideoData from "$lib/services/youtube";
+  import type { Thumbnails } from "$lib/services/server/youtube";
+  import getYoutubeVideoIDFromURL from "$lib/util/getYoutubeVideoIDFromURL";
   import YoutubeSubscribeLink from "$lib/components/YoutubeSubscribeLink";
+  import Image, { type Sources, type Srcset } from "$lib/components/Image";
 
   type VideoCardVariation = "hero" | "primary" | "secondary" | "tertiary";
   interface $$Props {
     class?: string;
     url: string;
-    customTitle?: string | null | undefined;
-    customDescription?: string | null | undefined;
+    title?: string | null | undefined;
+    description?: string | null | undefined;
     variation?: VideoCardVariation;
+    blurhash?: string | null | undefined;
+    thumbnails?: Thumbnails;
+    sizeType?: SizeType;
   }
 
   let className = "";
   export { className as class };
   export let url = "";
-  export let customTitle: $$Props["customTitle"] = null;
-  export let customDescription: $$Props["customDescription"] = null;
+  export let title: $$Props["title"] = null;
+  export let description: $$Props["description"] = null;
   export let variation = "primary";
 
-  let youtubeVideoId: string | null;
+  const maxDescriptionLength = 500;
+  $: trimmedDescription = description?.slice(0, maxDescriptionLength).replace(/\s[^\s]*?$/, "");
```

Comment:
Instead of stripping out newlines, I think we should remove everything after the first newline and just return the first paragraph. That way we can usually render full sentences, without the need to overflow part-way. The 500 character cut-off is fine but FWIW we accept a maximum of 256 characters for the `videoSubhead` field [in Contentful](https://app.contentful.com/spaces/pc5e1rlgfrov/content_types/videoWrapper/fields).

Reply:
This isn't actually stripping newlines, it's stripping "everything after and including the last whitespace" so that we don't cut things off in the middle of a word. I think we should keep this logic in case the first paragraph is too long, but I'll add your suggestion to cut it off at the first newline if that's shorter.